miscnn/processing/patching/patch_handler.py

```diff
@@ -0,0 +1,232 @@
+#==============================================================================#
+#  Author:       Philip Meyer                                                  #
+#  Copyright:    2022 IT-Infrastructure for Translational Medical Research,    #
+#                University of Augsburg                                        #
+#                                                                              #
+#  This program is free software: you can redistribute it and/or modify        #
+#  it under the terms of the GNU General Public License as published by        #
+#  the Free Software Foundation, either version 3 of the License, or           #
+#  (at your option) any later version.                                         #
+#                                                                              #
+#  This program is distributed in the hope that it will be useful,             #
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              #
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               #
+#  GNU General Public License for more details.                                #
+#                                                                              #
+#  You should have received a copy of the GNU General Public License           #
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.       #
+#==============================================================================#
+
+from .patch_operations import slice_matrix, concat_matrices, pad_patch, crop_patch
+import numpy as np
+from ..data_augmentation import Data_Augmentation
+
+
+class Patch_Handler():
```

Comment:
extra comment in general review

Reply:
Its main purpose is to split the ability to fit pictures for the network separate from the rest of the preprocessor to use with the image handling of XAI.

Reply:
It also makes the handling of properties cleaner.  In the preproessor some were static properties while some were passed in the constructor. This unifies all patching variables in one data object.

Reply:
Down the line i hope that this split allows for a simpler handling of cropping the borders of patches or handle smooth merging etc.

Reply:
Accepted